Heroes.ReplayParser/MPQFiles/ReplayMessageEvents.cs

```diff
@@ -41,23 +41,31 @@ public static void Parse(Replay replay, byte[] buffer)
                     {
                         case MessageEventType.SChatMessage:
                             {
-                                ChatMessage chatMessage = new ChatMessage();
+                                var chatMessage = new ChatMessage
+                                {
+                                        MessageTarget = (MessageTarget) bitReader.Read(3), Message = Encoding.UTF8.GetString(bitReader.ReadBlobPrecededWithLength(11))
+                                };
 
-                                chatMessage.MessageTarget = (MessageTarget)bitReader.Read(3); // m_recipient (the target)
-                                chatMessage.Message = Encoding.UTF8.GetString(bitReader.ReadBlobPrecededWithLength(11)); // m_string
+                                // m_recipient (the target)
+                                // m_string
 
                                 message.ChatMessage = chatMessage;
                                 replay.Messages.Add(message);
                                 break;
                             }
                         case MessageEventType.SPingMessage:
                             {
-                                PingMessage pingMessage = new PingMessage();
+                                var pingMessage = new PingMessage
+                                {
+                                        MessageTarget = (MessageTarget) bitReader.Read(3),
+                                        XCoordinate = bitReader.ReadInt32() - (-2147483648),
+                                        YCoordinate = bitReader.ReadInt32() - (-2147483648)
+                                };
 
-                                pingMessage.MessageTarget = (MessageTarget)bitReader.Read(3); // m_recipient (the target) 
+                                // m_recipient (the target) 
```

Comment:
Could you move these comments up to the statements they were originally associated with?